lib/browser-tests/puppeteer-helpers.js

```diff
@@ -0,0 +1,67 @@
+import path from 'path';
+import fs from 'fs';
+import { version } from '../../package.json';
+
+export function delay(ms) {
+  return new Promise((resolve) => {
+    setTimeout(resolve, ms);
+  });
+}
+
+const pathToSinonjsBrowserPkgSrc = path.resolve(__dirname, '../../node_modules/sinon/pkg/sinon.js');
+
+export async function useFakeTimers(page, config = {}) {
+  await page.evaluateOnNewDocument(`
+    if (!window.clock) {
+      ${fs.readFileSync(pathToSinonjsBrowserPkgSrc, 'utf8')}
+      window.clock = sinon.useFakeTimers(${JSON.stringify(config)})
+    }
+  `);
+}
+
+export async function clockTick(page, ms) {
+  // eslint-disable-next-line no-shadow
+  await page.evaluate((ms) => window.clock.tick(ms), ms);
+}
+
+export async function clockTickSeconds(page, s) {
+  await clockTick(page, s * 1000);
+}
+
+const preloadGovukFonts = fs.readdirSync(
```

Comment:
Is this so that the test is not slowed / flaky due to slow font loading?

Reply:
yup, for some of the timeout dialog tests there isn't any text rendered using the gov fonts on page load, only when the dialog triggers, and I guess css fonts are deferred by default so that's when the request is made to load them. Some of the timeout dialog tests intercept timeout / sign-out request to simulate a slow response and it was triggering on the fonts, so I copied over the preload code from the example app that's used for visual regression tests where we were getting some screenshots without fonts https://github.com/hmrc/hmrc-frontend/blob/5d05b9cb9745d51a088140277b16734206dad4cb/app/views/layouts/includes/head.html#L7